nim/app.py

```diff
@@ -34,12 +36,15 @@ def update_game_state():
 
 @app.route('/new', methods=['GET', 'POST'])
 def new_game():
-    json = loads('{"min": 3, "max": 12, "piles": 3}')
-    state = app.game.new_game(
-        min=json['min'],
-        max=json['max'],
-        piles=json['piles'],
-    )
+    game_request = request.get_json()
+    try:
+        state = app.game.new_game(
+            min=game_request['min'],
+            max=game_request['max'],
+            piles=game_request['piles'],
+        )
+    except NimException as e:
+        return jsonify({'error': e.args})
```

Comment:
We have to tell the front end something went wrong. I'd tell them that they made a bad request. The code for that is 400. Here's how you change your code from the default 200 to a 400.
`return jsonify({'error': e.args}), 400`

Reply:
Added, as you saw :D